File: internal/content/testdata/bench/java/sort-zeros-ones-twos.java
import java.util.*;

class Solution {
    public long[] sort_012(long[] nums) {
        long[] arr = Arrays.copyOf(nums, nums.length);
        int low = 0, mid = 0, high = arr.length - 1;
        while (mid <= high) {
            if (arr[mid] == 0) {
                long tmp = arr[low]; arr[low] = arr[mid]; arr[mid] = tmp;
                low++; mid++;
            } else if (arr[mid] == 1) {
                mid++;
            } else {
                long tmp = arr[mid]; arr[mid] = arr[high]; arr[high] = tmp;
                high--;
            }
        }
        return arr;
    }
}

File: internal/content/testdata/bench/go/sort-zeros-ones-twos.go
func sort_012(nums []int) []int {
	nums = append([]int{}, nums...)
	low, mid, high := 0, 0, len(nums)-1
	for mid <= high {
		if nums[mid] == 0 {
			nums[low], nums[mid] = nums[mid], nums[low]
			low++
			mid++
		} else if nums[mid] == 1 {
			mid++
		} else {
			nums[mid], nums[high] = nums[high], nums[mid]
			high--
		}
	}
	return nums
}
